arelle/DisclosureSystem.py

```diff
@@ -335,6 +341,45 @@ def loadStandardTaxonomiesDict(self):
                                                  level=logging.ERROR)
                 etree.clear_error_log()
 
+    def loadValidTaxonomiesDict(self):
+        if self.selection:
+            self.validTaxonomiesDict = defaultdict(set)
```

Comment:
This is probably pre-existing from the above method, but I think having `validTaxonomiesDict` be a `defaultdict` could cause issues. Some other code might unknowingly add a key to the dictionary just by inspecting it.

Reply:
Agree, would rather try and fix that in a different PR than this however.

Reply:
I can't find any place, outside of DisclosureSystem.py standardTaxonomiesDict is used as a dict, just a a set with in.  Suggest changing the externally referenced standardTaxonomiesDict into an frozen set (so no user messes with contents).

Reply:
Can't there be one method used for both loadStandardTaxonomiesDict and loadValidTaxonomiesDict, passing in the resulting  output dict or something unduplicative?